src/bif/map.rs

```diff
@@ -70,7 +92,19 @@ pub fn bif_maps_keys_1(_vm: &vm::Machine, process: &RcProcess, args: &[Value]) -
 }
 
 pub fn bif_maps_merge_2(_vm: &vm::Machine, _process: &RcProcess, args: &[Value]) -> BifResult {
-    unimplemented!();
+    let map1 = match &args[0] {
+        Value::Map(map) => &(*map.0),
+        _ => return Err(Exception::with_value(Reason::EXC_BADMAP, args[0].clone()))
+    };
+    let map2 = match &args[1] {
+        Value::Map(map) => &(*map.0),
+        _ => return Err(Exception::with_value(Reason::EXC_BADMAP, args[1].clone()))
+    };
+    let mut new_map = map2.clone();
+    for (k, v) in map1.iter() {
+        new_map = new_map.clone().plus(k.clone(), v.clone());
```

Comment:
I think since you cloned map2, you don't need to clone new_map inside this for loop since we own it by value.